dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java

```diff
@@ -239,6 +239,13 @@ public void provisionInputFiles(String parameterFilePath, List<Pair<String, Path
         executorService.shutdownNow();
     }
 
+    protected static URI createURIFromUnencodedPath(String filepath) {
+        //#1663
+        //pre-encoded paths will also work
+        URI fileIdentifier = URI.create(filepath.replace(" ", "%20"));
```

Comment:
We already import Guava which looks like it has a library method for this https://stackoverflow.com/revisions/31595036/6

Reply:
https://google.github.io/guava/releases/23.0/api/docs/com/google/common/net/UrlEscapers.html
any particular escaper? Seems like they each have specific character encodings

Reply:
The fragment escaper sounds appropriate and if it passes your tests, that's some additional confidence